test/catalyst/fixtures.ts

```diff
@@ -0,0 +1,101 @@
+import {ethers, deployments, getNamedAccounts} from 'hardhat';
+import {Contract, BigNumber} from 'ethers';
+
+const exampleGemId = 6;
+const notInOrderGemId = 56;
+const exampleCatalystId = 5;
+
+export const setupGemsAndCatalysts = deployments.createFixture(async () => {
+  await deployments.fixture();
+  const gemsCatalystsRegistry: Contract = await ethers.getContract(
+    'GemsCatalystsRegistry'
+  );
+  const powerGem: Contract = await ethers.getContract('Gem_Power');
+  const defenseGem: Contract = await ethers.getContract('Gem_Defense');
+  const speedGem: Contract = await ethers.getContract('Gem_Speed');
+  const magicGem: Contract = await ethers.getContract('Gem_Magic');
+  const luckGem: Contract = await ethers.getContract('Gem_Luck');
+  const commonCatalyst: Contract = await ethers.getContract('Catalyst_Common');
+  const rareCatalyst: Contract = await ethers.getContract('Catalyst_Rare');
+  const accounts = await getNamedAccounts();
+
+  await deployments.deploy(`Gem_Example`, {
+    contract: 'Gem',
+    from: accounts.deployer,
+    log: true,
+    args: ['Gem_Example', 'Gem_Example', accounts.deployer, exampleGemId],
+  });
+  const gemExample: Contract = await ethers.getContract('Gem_Example');
+
+  await deployments.deploy(`Gem_NotInOrder`, {
+    contract: 'Gem',
+    from: accounts.deployer,
+    log: true,
+    args: [
+      'Gem_NotInOrder',
+      'Gem_NotInOrder',
+      accounts.deployer,
+      notInOrderGemId,
+    ],
+  });
+  const gemNotInOrder: Contract = await ethers.getContract('Gem_NotInOrder');
+
+  await deployments.deploy(`Catalyst_Example`, {
+    contract: 'PrimaryCatalyst',
+    from: accounts.deployer,
+    log: true,
+    args: [
+      'Catalyst_Example',
+      'Catalyst_Example',
+      accounts.deployer,
+      5,
+      exampleCatalystId,
+    ],
+  });
+  const catalystExample: Contract = await ethers.getContract(
+    'Catalyst_Example'
+  );
+
+  await commonCatalyst
+    .connect(ethers.provider.getSigner(accounts.deployer))
+    .mint(accounts.deployer, BigNumber.from('8'));
```

Comment:
Same here A best practise is to use different account from deployer
I would have a special user account `catalystOwner` for example
that I would return as part of the fixture